File: JavaScript/Codility/GenomicRangeQuery.js
function solution(S, P, Q) {
    var answer = [];

    for (var i=0; i < P.length; i++){
        var tmpS = S.slice(P[i], Q[i]+1);

        if (tmpS.indexOf('A') !== -1){
            answer.push(1);
        }
        else if (tmpS.indexOf('C') !== -1){
            answer.push(2);
        }
        else if (tmpS.indexOf('G') !== -1){
            answer.push(3);
        }
        else {
            answer.push(4);
        }
    }
    return answer;
}

File: Python/Codility/GenomicRangeQuery.py
def solution(S, P, Q):
    answer = []
    for i in range(len(P)):
        tmpS = S[P[i]:Q[i]+1]
        if 'A' in tmpS:
            answer.append(1)
        elif 'C' in tmpS:
            answer.append(2)
        elif 'G' in tmpS:
            answer.append(3)
        else:
            answer.append(4)
    return answer